core/consensus/grandpa/impl/voting_round_impl.cpp

```diff
@@ -188,6 +188,10 @@ namespace kagome::consensus::grandpa {
     }
   }
 
+  VotingRoundImpl::~VotingRoundImpl() {
+    saveHistoricalVotes();
```

Comment:
I have a doubt about possible rewriting previously saved historical votes at destruction of temporary created round

Reply:
We must save our own vote (avoid equivocation).

I thought that calling `saveHistoricalVotes` for each `onVote` will perform poorly with batches (justification, catch-up).
Rust code saves `HistoricalVotes` for "concluded" and "completed" rounds,
which correspond to `doFinalize` and destruction of `GrandpaImpl::current_round_`.

Do you want to defer/merge writes?
```
struct VotingRoundImpl
  fn onVote(vote)
    history.push(vote)
    deferSave()
  fn deferSave
    dirty = true
    io.post {
      if dirty == true {
        dirty = false
        saveHistoricalVotes()
      }
    }
```

Reply:
Thanks for explanation:
```
onVote(1) // db = [1]
onCommit(2 3) // db = [2 3], 1 is lost
```